engine/signer.go

```diff
@@ -0,0 +1,146 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+
+	appchainClient "github.com/cosmos/cosmos-sdk/client"
+	cosmostx "github.com/cosmos/cosmos-sdk/client/tx"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+)
+
+type signer struct {
```

Comment:
Maybe: `signer` could live in its own package as a child of `engine`. Would make this struct a bit safer.

Reply:
Up to you! I think both approaches make sense

Reply:
Went ahead and did this. 64b17f8, 6c9e5b7.